File: src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultArticle.java
package com.pengrad.telegrambot.model.request;

/**
 * stas
 * 1/12/16.
 */
public class InlineQueryResultArticle extends InlineQueryResult {

    private String type;
    private String id;
    private String title;
    private String message_text;

    private ParseMode parse_mode;
    private Boolean disable_web_page_preview;
    private String url;
    private Boolean hide_url;
    private String description;
    private String thumb_url;
    private Integer thumb_width;
    private Integer thumb_height;

    public InlineQueryResultArticle(String id, String title, String messageText) {
        this.type = "article";
        this.id = id;
        this.title = title;
        this.message_text = messageText;
    }

    public InlineQueryResultArticle parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode;
        return this;
    }

    public InlineQueryResultArticle disableWebPagePreview(Boolean disableWebPagePreview) {
        this.disable_web_page_preview = disableWebPagePreview;
        return this;
    }

    public InlineQueryResultArticle url(String url) {
        this.url = url;
        return this;
    }

    public InlineQueryResultArticle hideUrl(Boolean hideUrl) {
        this.hide_url = hideUrl;
        return this;
    }

    public InlineQueryResultArticle description(String description) {
        this.description = description;
        return this;
    }

    public InlineQueryResultArticle thumbUrl(String thumbUrl) {
        this.thumb_url = thumbUrl;
        return this;
    }

    public InlineQueryResultArticle thumbWidth(Integer thumbWidth) {
        this.thumb_width = thumbWidth;
        return this;
    }

    public InlineQueryResultArticle thumbHeight(Integer thumbHeight) {
        this.thumb_height = thumbHeight;
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultArticle.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * stas
 * 1/12/16.
 */
class InlineQueryResultArticle(id: String, private val title: String, inputMessageContent: InputMessageContent) : InlineQueryResult<InlineQueryResultArticle>("article", id), Serializable {

    private var url: String? = null
    private var hide_url: Boolean? = null
    private var description: String? = null
    private var thumb_url: String? = null
    private var thumb_width: Int? = null
    private var thumb_height: Int? = null

    constructor(id: String, title: String, messageText: String) : this(id, title, InputTextMessageContent(messageText)) {}

    init {
        inputMessageContent(inputMessageContent)
    }

    fun url(url: String): InlineQueryResultArticle {
        this.url = url
        return this
    }

    fun hideUrl(hideUrl: Boolean?): InlineQueryResultArticle {
        this.hide_url = hideUrl
        return this
    }

    fun description(description: String): InlineQueryResultArticle {
        this.description = description
        return this
    }

    fun thumbUrl(thumbUrl: String): InlineQueryResultArticle {
        this.thumb_url = thumbUrl
        return this
    }

    fun thumbWidth(thumbWidth: Int?): InlineQueryResultArticle {
        this.thumb_width = thumbWidth
        return this
    }

    fun thumbHeight(thumbHeight: Int?): InlineQueryResultArticle {
        this.thumb_height = thumbHeight
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
